rust_snuba/src/processors/eap_spans.rs

```diff
@@ -119,32 +115,71 @@ impl From<FromSpanMessage> for EAPSpan {
                 &mut res.attr_str_~N,
                 )*
             ];
-            });
-
-            sentry_tags.iter().chain(tags.iter()).for_each(|(k, v)| {
-                attr_str_buckets[(fnv_1a(k.as_bytes()) as usize) % attr_str_buckets.len()]
-                    .insert(k.clone(), v.clone());
-            });
-        }
-
-        {
-            seq!(N in 0..20 {
             let mut attr_num_buckets = [
                 #(
                 &mut res.attr_num_~N,
                 )*
             ];
+            });
+
+            let mut insert_string = |k: String, v: String| {
+                attr_str_buckets[(fnv_1a(k.as_bytes()) as usize) % attr_str_buckets.len()]
+                    .insert(k.clone(), v.clone());
+            };
+
+            let mut insert_num = |k: String, v: f64| {
+                attr_num_buckets[(fnv_1a(k.as_bytes()) as usize) % attr_num_buckets.len()]
+                    .insert(k.clone(), v);
+            };
+
+            if let Some(sentry_tags) = from.sentry_tags {
+                sentry_tags.iter().for_each(|(k, v)| {
+                    if k == "transaction" {
```

Comment:
Why don’t we get this outside of the loop since it’s a map?

Reply:
otherwise it shows up twice - once in the bucket, and again normalized